### DISCUSSION TITLE
Plot color contours/discrete points in the spherical triangle using MTeX

### DISCUSSION THREAD

**Discussion by SagarChandra** [mtex-toolbox/mtex#675] [Ask Anything] [answered]
Hi all,

Is it possible to plot certain properties of grain boundaries, like energy, as color contours in the fundamental sectors like inverse pole figures? 
I have GB energy for various boundaries with different misorientation axis and boundary normals. I want to represent this data as color plots in the triangular sectors of misorientation space like inverse pole figures/fundamental zones.

Best regards,
Sagar

**Comment by kilir** [comment, marked as answer]
Hi,
yes, have a look at the help for `vector3d/plot` as well as `vector3d/interp`:

Example:
```
% random directions
v = vector3d.rand(1000)

% contour plot of some property of these vectors e.g. their inclination theta
plot(v,v.theta,'contour')
```

Hope that helps.

Cheers,
Rüdiger

**Comment by SagarChandra** [comment]
Hi Rüdiger,

Thanks for your reply. I looked into your suggestions; however, I have
distinct Miller indices which I want to plot into the spherical triangle
and color code each data point based on some scalar values, like grain
boundary energy. Is it still possible using vector.3d?

I am attaching a sample image of what I want to obtain.

Regards,
Sagar,

Thanks for your reply. I looked into your suggestions; however, I have
distinct Miller indices which I want to plot into the spherical triangle
and color code each data point based on some scalar values, like grain
boundary energy. Is it still possible using vector.3d?

I am attaching a sample image of what I want to obtain.

Regards,
Sagar

**Reply by kilir** [reply]
Hi,
I thought you requested contours. Anyhow, please have a look at `help vector3d/plot`, second argument is the data you want to  plot. vector3d or Miller does not really matter in that case.
Cheers,
Rüdiger

**Comment by SagarChandra** [comment]
Hi Rüdiger,

Thanks for your clarification. The issue is resolved now.

Regards.

**Reply by XiaoYinZ** [reply]
Hi SagarChandra,
I have some crystal directions(Euler angle), and a corresponding scalar value associated with each direction. I now want to visualize this data as filled contours in IPF map. I use "plotIPDF(ori,value,vector3d.X,'contourf');colorbar" to finish it, but there are no lines. When i use"plotIPDF(ori,value,vector3d.X);colorbar", figure shows points and corresponding value. So, how can i visualize corresponding scalar value as filled contours?
Thanks a lot!

**Reply by SagarChandra** [reply]
Hi Xiao,

Define ori = orientation.byEuler using the appropriate syntax

Then simply

plotIPDF(ori,a,vector3d.X,'smooth')

should do. Here, a is your data array.  

If you find error in interpolation, then see https://github.com/mtex-toolbox/mtex/commit/c52d339d6e30c4d7a3c51ebd6e0faee9c7d369cc.

HTH.
Sagar
